skiko/src/iosMain/kotlin/org/jetbrains/skiko/redrawer/MetalRedrawer.ios.kt

```diff
@@ -83,14 +103,15 @@ internal class MetalRedrawer(
 
     private fun draw() {
         // TODO: maybe make flush async as in JVM version.
-        autoreleasepool {
+        autoreleasepool { //todo maybe autoreleasepool is redundant, make measurements and talk with Kotlin Native team
             if (!isDisposed) {
                 contextHandler.draw()
             }
         }
```

Comment:
Need advice here

Reply:
One of the reasons why `autoreleasepool` is still needed nowadays is
> If you write a loop that creates many temporary objects

But it only applies to Objective-C objects that marked as "autorelease". However since it's hard to track what exactly will be allocated (ideally re-render the frame should have zero allocations if nothing changed), it's safe to keep it around drawing code.

Reply:
Yeah, thanks! I think so too.
Also, I asked this question to the Kotlin Native team.

Reply:
And inside `contextHandler.draw()` we have a lot of allocations and possible ObjectiveC interop.

Reply:
The Kotlin Native team advised to leave it as is. But measure performance.